go/vt/vttablet/tabletserver/planbuilder/dml.go

```diff
@@ -50,6 +50,12 @@ func analyzeUpdate(upd *sqlparser.Update, tables map[string]*schema.Table) (plan
 	}
 	table, tableErr := plan.setTable(tableName, tables)
 
+	// Updates aren't supported on topics
+	if tableErr == nil && table.IsTopic() {
+		plan.Reason = ReasonTopic
```

Comment:
Have all cases other than insert return an error instead. Not much point in delaying the error till executing time.